docs/DeveloperGuide.md

```diff
@@ -1,34 +1,511 @@
+
+
+---
+
+## Implementation
+
+This section describes some noteworthy details on how certain features are implemented.
+
+### User Interface
+The user interface of Plan&Score uses the singleton design. There is only one instance of UserInterface to be used throughout the application. The API for the user interface is UserInterface.java.
+
+#### Printing to user
+Instead of using the default method `System.out.println()` provided by Java to display messages to the user, use `showToUser(String …)` where multiple strings can be added into the arguments. Each string provided that is separated by `,` will be printed on the next line.
+
+Example input:
+```
+UserInterface userInterface = UserInterface.getInstance();
+
+String string1 = "Hello";
+String string2 = "How are you?";
+
+userInterface.showToUser(string1, string2);
+```
+
+Example output:
+```
+Hello
+How are you?
+```
+
+Code:
+```
+public void showToUser(String... message) {
+   for (String m : message) {
+       out.println(m);
+   }
+}
+```
+
+
+#### Printing arrays to user
+
+The method `printArray(ArrayList<String> stringArrayList)` is provided for the printing of any arraylists.
+
+Example input:
+```
+ArrayList<String> stringArrayList = new ArrayList<>();
+
+// Adding items into the stringArrayList
+stringArrayList.add("Hello");
+stringArrayList.add("How are you?);
+
+// Obtaining user interface instance
+UserInterface userInterface = UserInterface.getInstance();
+userInterface.printArray(stringArrayList);
+```
+
+Example output:
+```
+Hello
+How are you?
+```
+
+Code:
+```
+public void printArray(ArrayList<String> stringArrayList) {
+   assert stringArrayList != null;
+   for (String line : stringArrayList) {
+       userInterface.showToUser(line);
+   }
+}
+```
+
+### Help feature
+
+![helpcomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/Help.png)
+
+Figure 12. Sequence Diagram of the help feature
+
+When a user enters ‘help’, the input will be read in by the UI class.
+The UI class will then parse the user input into the ControlManager class, which calls the runLogic() method.
+The extractCommand() method of the CommandParser class is then called, extracting and returning the command type based on the user’s input.
+In this case, the command type would be ‘help’.
+The corresponding actionableCommand will be generated via the generateActionableCommand() method in the CommandFactory class.
+Lastly, the execute() method in the HelpCommand class is called, which in turn calls its own handleHelp() method.
+This displays the help message via the showToUser() method of the userInterface.
+
+Upon completion of this feature, it returns a boolean value “true” to the active flag in UserInterface.java to allow the continuous usage of the program.
+
+
+### Add feature
+
+![addcomponent](https://raw.githubusercontent.com/AY2021S1-CS2113T-W12-4/tp/master/docs/diagram/Add.png)
```

Comment:
i could not find DataManager class in your code, i only see EventDataManager. is the naming of the class in the diagram correct?